contracts/verifiers/UniversalVerifier.sol

```diff
@@ -57,117 +83,114 @@ contract UniversalVerifier is
         __Ownable_init(owner);
-        __ZKPVerifierBase_init(state);
+        __Verifier_init(state);
     }
 
     /// @dev Version of contract getter
     function version() public pure returns (string memory) {
         return VERSION;
     }
 
-    /// @dev Sets a ZKP request
-    /// @param requestId The ID of the ZKP request
-    /// @param request The ZKP request data
-    function setZKPRequest(
-        uint64 requestId,
-        IZKPVerifier.ZKPRequest calldata request
-    ) public override(RequestOwnership, ValidatorWhitelist, ZKPVerifierBase) {
-        super.setZKPRequest(requestId, request);
-
-        emit ZKPRequestSet(
-            requestId,
-            _msgSender(),
-            request.metadata,
-            address(request.validator),
-            request.data
-        );
+    /**
+     * @dev Sets an auth type
+     * @param authType The auth type to add
+     */
+    function setAuthType(IVerifier.AuthType calldata authType) public override onlyOwner {
+        super.setAuthType(authType);
+        emit AuthTypeSet(authType.authType, address(authType.validator), authType.params);
     }
 
     /**
-     * @dev Set the list of ZKP requests for the list of requestIds in the same order.
-     * @param requestIds Request ids of the ZKP requests.
-     * @param requests ZKP requests to set.
+     * @dev Sets different requests
+     * @param singleRequests The requests that are not in any group
+     * @param groupedRequests The requests that are in a group
      */
-    function setZKPRequests(
-        uint64[] calldata requestIds,
-        ZKPRequest[] calldata requests
-    ) public override(RequestOwnership, ValidatorWhitelist, ZKPVerifierBase) {
-        for (uint256 i = 0; i < requestIds.length; i++) {
-            setZKPRequest(requestIds[i], requests[i]);
+    function setRequests(
+        IVerifier.Request[] calldata singleRequests,
+        IVerifier.GroupedRequests[] calldata groupedRequests
+    ) public override(RequestOwnership, ValidatorWhitelist, Verifier) {
+        super.setRequests(singleRequests, groupedRequests);
+
+        for (uint256 i = 0; i < singleRequests.length; i++) {
+            emit RequestSet(
+                singleRequests[i].requestId,
+                _msgSender(),
+                singleRequests[i].metadata,
+                address(singleRequests[i].validator),
+                singleRequests[i].params
+            );
+        }
+
+        for (uint256 i = 0; i < groupedRequests.length; i++) {
+            for (uint256 j = 0; j < groupedRequests[i].requests.length; j++) {
+                emit RequestSet(
+                    groupedRequests[i].requests[j].requestId,
+                    _msgSender(),
+                    groupedRequests[i].requests[j].metadata,
+                    address(groupedRequests[i].requests[j].validator),
+                    groupedRequests[i].requests[j].params
+                );
+            }
         }
```

Comment:
The same, try to override the _setRequest method itself with additional functionality